spec/controllers/api/v0/products_controller_spec.rb

```diff
@@ -183,6 +184,27 @@
         expect(response.status).to eq(201)
         expect(json_response['name']).to eq("COPY OF #{product_with_image.name}")
       end
+
+      # test cases related to bug #660 - product duplication does not clone:
+      
+      # stock info
+      it '(does not) clone the stock info of the product' do
+        spree_post :clone, product_id: product.id, format: :json
+        expect(json_response['on_hand']).to eq(0)
+      end
+
+      # variants: "product" has 2 variants; the clone has 1 variant)
+      it '(does not) clone variants from a product with several variants' do
+        spree_post :clone, product_id: product.id, format: :json
+        expect(Spree::Product.second.variants.count).not_to eq Spree::Product.first.variants.count
+      end
+
+      #price info: it does not consider price changes
+      it '(does not) clone price' do
+        product.update_attribute(:price, 2.22)
+        spree_post :clone, product_id: product.id, format: :json
+        expect(json_response['price']).not_to eq(2.22)
```

Comment:
It doesn't clone the price? That seems really strange to me! What price does it actually have, in this case..?

Reply:
It seems to clone the price from the master variant, for that product. So, if the price of product is changed, the initial price will be considered, and not the value after updating. See also [this comment](https://github.com/openfoodfoundation/openfoodnetwork/issues/660#issuecomment-137298238)